From 1fb62fa9602377cbc69fd35d91233577c4cf075a Mon Sep 17 00:00:00 2001
From: debasishbsws <debasishbsws.dev@gmail.com>
Date: Tue, 26 Nov 2024 11:48:36 +0000
Subject: [PATCH] Fix(build): remove signing_key from gemspec

Our ruby/build pipeline delete all the lines contains 'signing_key' by default since it leads to build failures as the key is not available in the build environment and is not needed. ref: https://github.com/chainguard-dev/melange/blob/main/pkg/build/pipelines/ruby/build.yaml#L42

But for version 2.0.3 and later it resutls in syntax error, unexpected end as it removes the condition line but not the full condition.

Signed-off-by: debasishbsws <debasishbsws.dev@gmail.com>
---
 logstash-output-opensearch.gemspec | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/logstash-output-opensearch.gemspec b/logstash-output-opensearch.gemspec
index cd04d97..1b0af3d 100644
--- a/logstash-output-opensearch.gemspec
+++ b/logstash-output-opensearch.gemspec
@@ -29,10 +29,7 @@ Gem::Specification.new do |s|
   # Tests
   s.test_files = s.files.grep(%r{^(test|spec|features)/})
 
-  if $PROGRAM_NAME.end_with?("gem") && ARGV == ["build", __FILE__] && File.exist?(signing_key_path)
-    s.signing_key = signing_key_path
-    s.cert_chain  = ['certs/opensearch-rubygems.pem']
-  end
+  s.cert_chain  = ['certs/opensearch-rubygems.pem']
 
   # Special flag to let us know this is actually a logstash plugin
   s.metadata = {
-- 
2.47.1

